### ISSUE TITLE
Race condition with "ssh -tt" prevents me from using {tty => 1}

### ISSUE THREAD

**Issue by hoytech** [salva/p5-Net-OpenSSH#19]
I don't believe this is an issue in Net::OpenSSH itself. This is more an FYI but I'd be interested in any input.

I'm trying to get sudo to work with `{tty => 1}` except I'm hitting what seems to be a race condition with OpenSSH. If anyone is interested I've made a test script to reproduce it here:

https://gist.github.com/hoytech/d3e19c6e860e224cbdd7

Because of this I've given up on `{tty => 1}` and am removing the "Defaults  requiretty" line from `/etc/sudoers`... Damn redhat/AWS... At least they are removing this config from their default in future releases:

https://bugzilla.redhat.com/show_bug.cgi?id=1020147

Sorry to keep bothering you :)

**Comment by salva** [comment]
It is a serve side issue. The server is abort the connection:

```
Feb  5 15:41:36 oscar sshd[3793]: Accepted publickey for salva from ::1 port 39824 ssh2: DSA SHA256:...
Feb  5 15:41:36 oscar sshd[3793]: pam_unix(sshd:session): session opened for user salva by (uid=0)
Feb  5 15:41:36 oscar systemd-logind[844]: New session 19 of user salva.
Feb  5 15:41:36 oscar sshd[3838]: fatal: mm_request_receive_expect: read: rtype 29 != type 157
Feb  5 15:41:36 oscar sshd[3837]: error: buffer_get_char_ret: incomplete message
Feb  5 15:41:36 oscar sshd[3837]: fatal: buffer_get_char: buffer error
Feb  5 15:41:36 oscar sshd[3793]: fatal: mm_request_receive: read: Connection reset by peer
Feb  5 15:41:36 oscar sshd[3793]: pam_unix(sshd:session): session closed for user salva
Feb  5 15:41:36 oscar systemd-logind[844]: Removed session 19.
```

I can reproduce it on Ubuntu but not on OpenBSD.

I am investigating it further...

**Comment by salva** [comment]
The bug is caused by an Ubuntu  specific patch!!!

https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/1334916

Will be fixed in Ubuntu 16.04.

**Comment by hoytech** [comment]
Thank you very much for tracking this down!
